src/App.tsx

```diff
@@ -2,23 +2,94 @@ import React from 'react';
 import './App.scss';
 
 const goodsFromServer: string[] = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+  'Sex',
+  'Drugs',
+  'Rock-n-Roll',
+  'Love',
+  'Passion',
+  'Alcohol',
+  'Party',
+  'Adventure',
+  'Laugh',
+  'Money',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: [],
+  };
+
+  toggler = (good: string) => {
+    if (good) {
+      this.setState((state) => {
+        return !this.alreadyInSelectedGoods(good)
+          ? { selectedGoods: [...state.selectedGoods, good] }
+          : { selectedGoods: state.selectedGoods.filter(item => item !== good) };
+      });
+    }
+  };
+
+  alreadyInSelectedGoods = (good: string) => {
+    return this.state.selectedGoods.find(item => item === good);
+  };
+
+  selectedGoodsInfoForUser = () => {
+    const { selectedGoods } = this.state;
+
+    switch (selectedGoods.length) {
+      case 0:
+        return 'Choose Things To Enjoy';
+      case 1:
+        return `${selectedGoods[0]} is selected`;
+      case 2:
+        return `${selectedGoods[0]} and ${selectedGoods[1]} are selected`;
+      default:
+        return `${selectedGoods.slice(0, -1).join(', ')} and ${selectedGoods.slice(-1)} are selected`;
+    }
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <div className="App__container">
+          <div className="App__top-section">
+            <h1 className="App__goods-info">
+              {this.selectedGoodsInfoForUser()}
+            </h1>
+            <button
+              type="button"
+              className="App__close-button"
+              onClick={() => {
+                this.setState({ selectedGoods: [] });
+              }}
+            >
+              XXX
+            </button>
+          </div>
+          <ul className="App__list">
+            {goodsFromServer.map((good) => {
+              return (
+                <li key={good} className={this.alreadyInSelectedGoods(good) ? 'list list-selected' : 'list'}>
```

Comment:
It's better to use classnames library

Reply:
@dariia-romanova I wanted to use it but in this particular case my choice seems more readable to me

Reply:
I vote for classnames too )